src/app/eth-common/module/search/component/SearchInline.tsx

```diff
@@ -56,20 +59,24 @@ const SearchBoxContainer = styled.div`
 `;
 
 export interface ISearchInlineProps {
-    internalNav: IInternalNav;
+    internalNav?: IInternalNav;
     translation: ITranslation;
     search: ISearch;
     searchInlineStore: SearchInlineStore;
+    logger: ILogger;
     onRequestClose?(): void;
 }
 
 @observer
 class $SearchInline extends React.Component<ISearchInlineProps> {
     private searchBox: HTMLInputElement;
-    @observable
-    private noResults = false;
-    @observable
-    private inProgress = false;
+    private searchState: SearchState;
+
+    constructor(props: ISearchInlineProps) {
+        super(props);
+
+        this.searchState = new SearchState(this.props.search, this.props.internalNav!, this.props.logger);
```

Comment:
Because I extracted the internal state of the component in a separate class. It shouldn't be instantiated or visible outside the component at all. It should be fully transparent.